plugins/user-authenticators/saml2/src/org/apache/cloudstack/saml/SAML2AuthManagerImpl.java

```diff
@@ -368,8 +368,14 @@ private boolean setup() {
                 _idpMetaDataProvider = new HTTPMetadataProvider(_timer, client, idpMetaDataUrl);
             } else {
                 File metadataFile = PropertiesUtil.findConfigFile(idpMetaDataUrl);
-                s_logger.debug("Provided Metadata is not a URL, trying to read metadata file from local path: " + metadataFile.getAbsolutePath());
-                _idpMetaDataProvider = new FilesystemMetadataProvider(_timer, metadataFile);
+                if (metadataFile == null) {
+                    s_logger.error("Metadata file returned null");
```

Comment:
Can you also add the file path its trying to fing (idpMetaDataUrl) to the error message?